src/pages/index.js

```diff
@@ -19,50 +21,136 @@ import {
   addNewCardButton,
   settings,
 } from "../utils/constants.js";
+import PopupWithConfirm from "../components/PopupWithConfirm.js";
+import Api from "../components/API.js";
+import { data } from "autoprefixer";
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "50644799-43da-42c8-ae19-a67fa2a948b2",
+    "Content-Type": "application/json",
+  },
+});
 
 const previewImagePopup = new PopupWithImage({
   popupSelector: "#card-preview-modal",
 });
-const addCardPopup = new PopupWithForm("#add-card-modal", (data) => {
-  const cardData = {
-    name: data.title,
-    link: data.url,
-  };
-  const newCard = createCard(cardData);
-  cardList.addItem(newCard);
-  addCardFormElement.reset();
-  addFormValidator.disableButton();
-});
 
 const cardList = new Section(
   {
-    items: initialCards,
+    items: [],
     renderer: (data) => {
       cardList.addItem(createCard(data));
     },
   },
   ".cards__list"
 );
 
+const confirmDeleteModal = new PopupWithConfirm(
+  "#delete-modal",
+  handleDeleteClick
+);
+confirmDeleteModal.setEventListeners();
+
+const editAvatarModal = new PopupWithForm("#edit-avatar-modal", (input) => {
+  editAvatarModal.setLoading("Saving...");
```

Comment:
`  editAvatarModal.setLoading("Saving...");
`

Please, place it inside `handleAvatarEditSubmit` to make the code consistent


And you need to add `finally` there to stop the loading effect